### PR TITLE
Dev/comment author invisible fix

### PR DESCRIPTION
**Pull request by demet47**
Added DTO classes to provide just the related info and add the missing info requested by frontend.

### CODE REVIEW

**Comment by demet47** [comment]
Following is what is returned from call to [GET] http://localhost:8080/art_item/13. Is it feasible @CahidArda since I also reduced some fields too.

```
{
    "name": "Starry Night",
    "description": "A reflection of my soul from a fuzzy night.",
    "category": "Classical",
    "imageUrl": null,
    "labels": [],
    "creatorAccountInfo": {
        "email": "email@email.com",
        "username": "janedoe",
        "id": 8,
        "name": "Jane",
        "surname": "Doe",
        "country": "England",
        "dateOfBirth": null,
        "profilePictureUrl": null
    },
    "creatorId": 15,
    "creationDate": "2022-11-14T19:14:44.000+00:00",
    "ownerAccountInfo": {
        "email": "email@email.com",
        "username": "janedoe",
        "id": 8,
        "name": "Jane",
        "surname": "Doe",
        "country": "England",
        "dateOfBirth": null,
        "profilePictureUrl": null
    },
    "ownerId": 7,
    "onAuction": false,
    "auction": null,
    "lastPrice": 0.0,
    "commentList": [
        {
            "id": 21,
            "text": "Wonderful! I feel goosebumps!",
            "creationDate": "2022-11-14T19:14:44.000+00:00",
            "lastEditDate": "2022-11-14T19:14:44.000+00:00",
            "upVotedUserIds": [],
            "downVotedUserIds": [],
            "reports": [],
            "authorAccountInfo": {
                "email": "email3@email.com",
                "username": "janettdoe",
                "id": 12,
                "name": "Janett",
                "surname": "Doe",
                "country": "Katalan",
                "dateOfBirth": null,
                "profilePictureUrl": null
            },
            "authorId": 11
        },
        {
            "id": 22,
            "text": "Amazing!",
            "creationDate": "2022-11-14T19:14:44.000+00:00",
            "lastEditDate": "2022-11-14T19:14:44.000+00:00",
            "upVotedUserIds": [],
            "downVotedUserIds": [],
            "reports": [],
            "authorAccountInfo": {
                "email": "email2@email.com",
                "username": "joedoe",
                "id": 10,
                "name": "Joe",
                "surname": "Doe",
                "country": "Sydney",
                "dateOfBirth": null,
                "profilePictureUrl": null
            },
            "authorId": 9
        }
    ],
    "bookMarkedByIds": []
}
```

**Comment by demet47** [comment]
@sabrimete @CahidArda  beware, I also changed the return tyoe of create art item with ArtItemDTO instead of ArtItem to provide a consistent return type. Please do request rollback for that if you think it will create any chain problems which I can't.

**Comment by sabrimete** [comment]
There is a problem with imageUrl field. 
Problem occurs because of the previous PRs. 
Probably you couldn't realize it because you merge the develop branch after you completed this PR. 

You should work on the current merged version to fix the issue.

**Review by CahidArda — APPROVED** [review]
As far as I undestand, @sabrimete was refering to an issue in CI. It looks like this is fixed now.

I haven't tested the endpoint locally but the response sample you provided solves the issue. Changes are fine by me. Thanks for the work.

**Comment by CahidArda** [comment]
Can we merge this PR if the problem @sabrimete pointed out has been resolved? We need this PR merged for adding proper comment sections.

**Comment by sabrimete** [comment]
> Can we merge this PR if the problem @sabrimete pointed out has been resolved? We need this PR merged for adding proper comment sections.

There are conflicts to be resolved right now. I will review the PR after they are solved.

**Comment by demet47** [comment]
> > Can we merge this PR if the problem @sabrimete pointed out has been resolved? We need this PR merged for adding proper comment sections.
> 
> There are conflicts to be resolved right now. I will review the PR after they are solved.

Done

**Comment by CahidArda** [comment]
Any updates on this PR? I am waiting for this to merge for updating the comment sections.

File: backend/src/main/kotlin/com/group7/artshare/controller/ArtItemController.kt
Comment (sabrimete) [inline backend/src/main/kotlin/com/group7/artshare/controller/ArtItemController.kt:37]: We can remove nullable return type by changing the code like this
I recommend not using methods like Objects.nonNull. It is better to use Kotlin way of null checking because if you do it other way, compiler may not understand that you checked. So it gives warning that says it can be null make the return type nullable etc.

**Review by sabrimete — APPROVED** [review]
I've made a comment about returning type. However, it is not a required change right now. Keep it in my mind for future implementations. 
Since other teams are waiting for this PR, I'm approving as it is.

**Pull request merged**
